### PR TITLE
update CSRF documentation

### PR DESCRIPTION
**Pull request by fusuiyi123**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [x] Have you referenced any issues you're fixing using [commit message keywords](https://help.github.com/articles/closing-issues-using-keywords/)?
* [x] Have you added copyright headers to new files?
* [x] Have you checked that both Scala and Java APIs are updated?
* [x] Have you updated the documentation for both Scala and Java sections?
* [x] Have you added tests for any changed functionality?

# Helpful things

## Fixes

Fixes https://github.com/playframework/playframework/issues/9187

## Purpose

Update CSRF documentation to clarify
>To allow simple protection for non browser requests, Play only checks requests with cookies in the header.

which should check `Cookie` or `Authorization` by default

### CODE REVIEW

File: documentation/manual/working/javaGuide/main/forms/JavaCsrf.md [hunk 1]
```diff
@@ -11,13 +11,13 @@ There is no simple answer to what requests are safe and what are vulnerable to C
...
 
 ### Play's CSRF protection
 
 Play supports multiple methods for verifying that a request is not a CSRF request.  The primary mechanism is a CSRF token.  This token gets placed either in the query string or body of every form submitted, and also gets placed in the users session.  Play then verifies that both tokens are present and match.
 
-To allow simple protection for non browser requests, Play only checks requests with cookies in the header.  If you are making requests with AJAX, you can place the CSRF token in the HTML page, and then add it to the request using the `Csrf-Token` header.
+To allow simple protection for non browser requests, Play only checks requests with `Cookie` or `Authorization` headers. You can configure `play.filters.csrf.header.protectHeaders` to defines headers that must be present to perform the CSRF check. If you are making requests with AJAX, you can place the CSRF token in the HTML page, and then add it to the request using the `Csrf-Token` header.
```
[the remark this answers is not in the document]
Comment (gmethvin) [reply documentation/manual/working/javaGuide/main/forms/JavaCsrf.md]: And it should be “to define”, not “to defines”

File: documentation/manual/working/scalaGuide/main/forms/ScalaCsrf.md [hunk 2]
```diff
@@ -11,13 +11,13 @@ There is no simple answer to what requests are safe and what are vulnerable to C
...
 
 ### Play's CSRF protection
 
 Play supports multiple methods for verifying that a request is not a CSRF request.  The primary mechanism is a CSRF token.  This token gets placed either in the query string or body of every form submitted, and also gets placed in the user's session.  Play then verifies that both tokens are present and match.
 
-To allow simple protection for non-browser requests, Play only checks requests with cookies in the header.  If you are making requests with AJAX, you can place the CSRF token in the HTML page, and then add it to the request using the `Csrf-Token` header.
+To allow simple protection for non-browser requests, Play only checks requests with `Cookie` or `Authorization` headers. You can configure `play.filters.csrf.header.protectHeaders` to defines headers that must be present to perform the CSRF check. If you are making requests with AJAX, you can place the CSRF token in the HTML page, and then add it to the request using the `Csrf-Token` header.
```
Comment (gmethvin) [inline documentation/manual/working/scalaGuide/main/forms/ScalaCsrf.md]: Same comments as above

**Comment by fusuiyi123** [comment]
Hi @gmethvin thanks for your comment, just addressed it:)

**Comment by mkurz** [comment]
@Mergifyio backport 2.8.x

**Pull request merged**